tests/tekton-resources/tasks/generators/clusterloader/load-pod-identity.yaml

```diff
@@ -175,6 +178,68 @@ spec:
       S3_RESULT_PATH=$(cat $(results.s3_result.path))
       echo "S3 Path: $S3_RESULT_PATH"
       aws sts get-caller-identity
+
+      REGION=$(params.region)
+      ENDPOINT_FLAG=""
+      if [ -n "$(params.endpoint)" ]; then
+        ENDPOINT_FLAG="--endpoint $(params.endpoint)"
+      fi
+
+      CLUSTER_NAME=$(params.cluster-name)
+      NAMESPACE="EKSPodIdentityScalabilityTests"
+      DIMENSION_NAME="ClusterName"
+      DIMENSION_VALUE=$CLUSTER_NAME
+      METRIC_LATENCY_NAME="CredentialFetchLatency"
+      PERIOD=300
+
+      START_TIME=$(aws eks $ENDPOINT_FLAG --region $REGION describe-cluster \
+        --name "$CLUSTER_NAME" \
+        --query "cluster.createdAt" \
+        --output text)
+
+      END_TIME=$(date -u +"%Y-%m-%dT%H:%M:%SZ")
+
+      response=$(aws cloudwatch get-metric-statistics \
+        --region "$REGION" \
+        --namespace "$NAMESPACE" \
+        --metric-name "$METRIC_LATENCY_NAME" \
+        --dimensions Name="$DIMENSION_NAME",Value="$DIMENSION_VALUE" \
+        --start-time "$START_TIME" \
+        --end-time "$END_TIME" \
+        --period "$PERIOD" \
+        --extended-statistics p50 p90 p99 \
+        --output json)
+
+      # extract p50 p90 p99
+      latest=$(echo "$response" | jq -r '.Datapoints | sort_by(.Timestamp) | last')
+      p50=$(echo "$latest" | jq -r '."ExtendedStatistics"."p50" // "N/A"')
+      p90=$(echo "$latest" | jq -r '."ExtendedStatistics"."p90" // "N/A"')
+      p99=$(echo "$latest" | jq -r '."ExtendedStatistics"."p99" // "N/A"')
+
+      response=$(aws cloudwatch get-metric-statistics \
+        --region "$REGION" \
+        --namespace "$NAMESPACE" \
+        --metric-name "$METRIC_LATENCY_NAME" \
+        --dimensions Name="$DIMENSION_NAME",Value="$DIMENSION_VALUE" \
+        --start-time "$START_TIME" \
+        --end-time "$END_TIME" \
+        --period "$PERIOD" \
+        --statistics SampleCount \
+        --output json)
+
+      total_samples=$(echo "$response" | jq '[.Datapoints[].SampleCount] | add // 0')
+
+      cat <<EOF > eks_pod_identity_test_summary.json
+      {
+        "start_time": "$START_TIME",
+        "end_time": "$END_TIME",
+        "total_samples": $total_samples,
+        "p50": $p50,
+        "p90": $p90,
+        "p99": $p99
```

Comment:
I see you are printing the latency numbers, but i don't see you validating the p99 under `x` seconds. Am i missing something here ?

Reply:
There are 2 options:
1. fail the task on threshold breaching
2. integrate with alarm on threshold breaching

Reply:
We have the metrics, and can integrate with an alarm, this should be out of this PR. I still think we should give a failure signal, just from cognitive perspective, a successful run is being alarmed, it can cause confusion